src/core/registration.c

```diff
@@ -52,6 +52,14 @@ MsQuicRegistrationOpen(
         goto Error;
     }
 
+    //
+    // Lazy initialize partitions.
+    //
+    Status = MsQuicLibraryInitializePartitions(ExternalRegistration);
```

Comment:
Passing this flag should not be necessary because external registrations are always created first. Instead, remove this check, and put this in the `if (ExternalRegistration)` below.